### PR TITLE
(master) Mx builds should use python3 to avoid issues

### LINKED ISSUE #208: Make mandrel-packaging set python version to be used with mx to 3
As python 2 reached its end of life in Jan 2020, we should make sure we no longer use it when building mandrel. `mx` doesn't seem to enforce the use of python3 so if python2 is found in the path it might as well use it.

Related issues:
* https://github.com/graalvm/mx/issues/247
* https://github.com/graalvm/mx/issues/238

### PR DESCRIPTION
**Pull request by galderz**
Closes #208

### CODE REVIEW

File: build.java [hunk 1]
```diff
@@ -903,6 +909,7 @@ static void build(
             return Tasks.Exec.of(
                 args
                 , mandrelRepo.resolve("substratevm")
+                , new EnvVar("MX_PYTHON", "python3")
```
Comment (zakkak) [inline build.java:912]: This only sets the env variable for `mxbuild`.

`mxclean`, `mxversion` and `mandrelVersion` will still run with python 2.

I suggest always adding the variable in `Tasks.Exec.envVars` (even for no mx commands)

**Review by zakkak — CHANGES_REQUESTED** [review]

File: build.java (hunk 1 above)
Comment (galderz) [reply build.java:912]: I don't agree with adding the env var to no mx commands. It serves no purpose. I will look into some way to add it to only the mx commands.

**Comment by galderz** [comment]
@zakkak I've updated the PR. I've made such that the env var is set for all mx commands. If you're happy with it I'll backport the latest changes to 21.3.

**Review by zakkak — APPROVED** [review]
Thanks @galderz 

LGTM

**Pull request merged**